app/assets/javascripts/visualizations/highvis/baseVis.coffee.erb

```diff
@@ -113,279 +115,258 @@ $ ->
-              #{if data.groupSelection.length == data.groups.length then "checked" else ""}>
-              #Check All</label></div>"""
-            controls += "</td><td></td></tr>"
-
-        for group, gIndex in data.groups
-          color_id = counter % globals.configs.colors.length
-          color = globals.configs.colors[color_id]
-          controls += "<tr><td class='label-dataset-name'>"
-          controls += "<div class='inner_control_div' id='label-color-#{color_id}'"
-          controls += "style=\"color:#{color};\">"
-
-          if radio
-            # Radio groups can only have one group selected
-            groupSelection = [data.groupSelection.reduce (a, b) -> Math.min(a,b)]
-            data.groupSelection = groupSelection
-
-            selectedGroup = data.groupSelection[0]
-            controls += "<div class='radio'><label><input class='group_input' type='radio' name='groups'"
-            controls += " value='#{gIndex}' "
-            controls += "#{if (Number gIndex) is selectedGroup then "checked" else ""}/>"
-            if group isnt ''
-              controls += "#{group}</label>"
-            else
-              controls +="No #{data.fields[globals.configs.groupById].fieldName}</label>"
-          else
-            controls += "<div class='checkbox'><label><input class='group_input' type='checkbox'"
-            controls += " value='#{gIndex}' "
-            controls += "#{if (Number gIndex) in data.groupSelection then "checked" else ""}/>"
-            if group isnt ''
-              controls += "#{group}</label>"
-            else
-              controls +="No #{data.fields[globals.configs.groupById].fieldName}</label>"
-          controls += "</div></div>"
-          controls += "</td><td class='label-dataset-color'>"
-          controls += """<a href="javascript:void;" class="color-picker" data-color="#{color}" data-color-id="#{color_id}">
-                         <img height="20px" width="20px" src="<%= image_path('color-wheel.png') %>"></a>"""
-          controls += "</td></tr>"
-          counter += 1
-
-        controls += "</table>"
-
-        controls += '</div></div>'
-
-        # Write HTML
-        ($ '#controldiv').append controls
-
-        ($ '.color-picker').each (_, cpicker) =>
-          cid = $(cpicker).attr 'data-color-id'
+        groupList = undefined
+        groupList =
+          for g, i in data.groups
+            color = globals.getColor(i)
+            label =
+              if g isnt '' then g
+              else "No #{data.fields[globals.configs.groupById].fieldName}"
+            style:   "color: #{color}; width: 90%"
+            value:   i
+            label:   label
+            name:    'grouping'
+            id:      'group-' + i
+            logId:   'group-' + i
+            picker:
+              type:  'color'
+              color: color
+              icon:  '<%= image_path('color-wheel.png') %>'
+
+        ictx =
```

Comment:
I like @JKinsman editing my comments and adding :+1: instead of commenting himself :goat: :laughing: